### DISCUSSION TITLE
Validation step: error when trying to return object

### DISCUSSION THREAD

**Discussion by mcomunita** [Lightning-AI/pytorch-lightning#11916] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi,

As part of my _validation_step_ in a AE I am trying to return an object containing input and reconstructed output.

_validation_epoch_end_ does not get the accumulated outputs.

Here's the code:
```
def validation_step(self, batch, batch_idx):
        x, x_recon, _, _, _ = self.forward(batch)
        outputs = {
            'x': x,
            'x_recon': x_recon
        }
        return outputs
```

these are the shapes:
`x.shape:  torch.Size([2, 1, 257, 63])`
`x_recon.shape:  torch.Size([2, 1, 257, 63])`

and this is what _validation_step_outputs_ looks like:
`[{'x': tensor(0.0033, device='cuda:0'), 'x_recon': tensor(-0.0102, device='cuda:0')}]`

Any idea why?

Thanks

**Comment by rohitgr7** [comment, marked as answer]
hey @mcomunita!

this shouldn't happen. Quick ques: are you using DP?

**Reply by mcomunita** [reply]
Hi @rohitgr7,

Yes, I am using DP. It took me a while to figure it out but I saw on the documentation that in such a case I need to also implement _validation_step_end_, which I am now using to accumulate the outputs from _validation_step_

You can consider this solved.

Thanks

**Reply by rohitgr7** [reply]
Also we recently patched this, so it won't reduce any other outputs except the loss. Will be available in the next release :)
